WordPress/gutenberg
# Ensure block toolbar doesn't overlap block by modifying forcePosition and shift popover props

## What?
Fixes #41575

Restores the previous behavior where the block toolbar flips below the block when there's not enough space at the top of the canvas.

## How?
Modifies the popover props on the fly. When there's enough space at the top of the canvas, the props `{ __unstableForcePosition: true, __unstableShift: true }` are used, disabling the toolbar 'flip' behavior and enabling the 'shift' behavior.

When there's a restricted amount of space, the props `{ __unstableForcePosition: false, __unstableShift: false }` are used, which enable the 'flip' behavior and disable the 'shift' behavior.

## Testing Instructions
1. Open the site editor, go to 'Templates' and choose Blank (which appears in the Twenty Twenty Two theme)
2. Insert some blocks, ideally enough for the page to be scrollable.
3. Select the first block—if there isn't enough space above the block it should appear below.

## Video

https://user-images.githubusercontent.com/677833/182343939-4d9987ae-97af-433f-bf4a-e71f7b1d94ac.mp4
